cli/dcoscli/service/main.py

```diff
@@ -123,3 +146,170 @@ def _shutdown(service_id):
+    :type file_: str
+    :returns: process return code
+    :rtype: int
+    """
+
+    dcos_client = mesos.DCOSClient()
+    task = mesos.get_master(dcos_client).task(task_id)
+    mesos_file = mesos.MesosFile(file_, task=task, dcos_client=dcos_client)
+    return log.log_files([mesos_file], follow, lines)
+
+
+def _get_service_task(service_name):
+    """Gets the task running the given service.  If there is more than one
+    such task, throws an exception.
+
+    :param service_name: service name
+    :type service_name: str
+    :returns: The marathon task dict
+    :rtype: dict
+    """
+
+    marathon_client = marathon.create_client()
+    app = _get_service_app(marathon_client, service_name)
+    tasks = marathon_client.get_app(app['id'])['tasks']
+    if len(tasks) != 1:
+        raise DCOSException(
+            ('We expected marathon app [{}] to be running 1 task, but we ' +
+             'instead found {} tasks').format(app['id'], len(tasks)))
+    return tasks[0]
+
+
+def _get_service_app(marathon_client, service_name):
+    """Gets the marathon app running the given service.  If there is not
+    exactly one such app, throws an exception.
+
+    :param marathon_client: marathon client
+    :type marathon_client: marathon.Client
+    :param service_name: service name
+    :type service_name: str
+    :returns: marathon app
+    :rtype: dict
+    """
+
+    apps = package.get_apps_for_framework(service_name, marathon_client)
+
+    if len(apps) > 1:
+        raise DCOSException(
+            'Multiple marathon apps found for service name [{}]: {}'.format(
+                service_name,
+                ', '.join('[{}]'.format(app['id']) for app in apps)))
+    elif len(apps) == 0:
+        raise DCOSException(
+            'No marathon apps found for service [{}]'.format(service_name))
+    else:
+        return apps[0]
+
+
+def _log_marathon(follow, lines, config_file):
+    """Prints the contents of the marathon logs.
+
+    :param follow: same as unix tail's -f
+    :type follow: bool
+    :param lines: number of lines to print
+    :type lines: int
+    :param config_file: SSH config file.
+    :type config_file: str | None
+    ;:returns: process return code
+    :rtype: int
+    """
+
+    error = ('`dcos service log marathon` runs ssh to '
+             'access the systemd journal on the leading marathon.  To enable '
+             'this, you must run `ssh-agent` so your private key can be '
+             'used safely to hop between nodes in your cluster.')
+    ssh_options = util.get_ssh_options(config_file, [], error)
+
+    journalctl_args = ''
+    if follow:
+        journalctl_args += '-f '
+    if lines:
+        journalctl_args += '-n {} '.format(lines)
+
+    leader_ip = marathon.create_client().get_leader().split(':')[0]
+    master_public_ip = mesos.DCOSClient().metadata()['PUBLIC_IPV4']
+    user = 'core'
+
+    cmd = ("ssh -A -t {3} {0}@{1} " +
+           "ssh -A -t {0}@{2} " +
```

Comment:
Let's come to an agreement on the -A option. Also, you are running `journalctl` so do you want the `-T` (disable pty) or `-t` (enable pty)?